bin/convert-blast-xml-to-json.py

```diff
@@ -1,23 +1,48 @@
 #!/usr/bin/env python
 
+import argparse
+import bz2
 import sys
-from os.path import basename
 
 from dark.blast.conversion import XMLRecordsReader
 
+
 if __name__ == '__main__':
-    nArgs = len(sys.argv)
-    if nArgs == 1:
-        reader = XMLRecordsReader(sys.stdin)
-        reader.saveAsJSON(sys.stdout)
-    elif nArgs == 2:
-        reader = XMLRecordsReader(sys.argv[1])
-        reader.saveAsJSON(sys.stdout)
-    elif nArgs == 3:
-        reader = XMLRecordsReader(sys.argv[1])
-        with open(sys.argv[2], 'w') as fp:
+
+    parser = argparse.ArgumentParser(
+        description='Convert a BLAST XML file to JSON.',
+        epilog='Give a BLAST XML file and convert it to JSON. Optionally '
+        'compress the JOSN output file.'
+    )
+
+    parser.add_argument(
+        'xml', metavar='BLAST-XML-file', type=str, help='the XML file of '
+        'BLAST output.')
+
+    parser.add_argument(
+        'json', metavar='BLAST-JSON-file', type=str, nargs='?',
+        help='the JSON filename where contents of "xml" should be written to.')
+
+    parser.add_argument(
+        '--compress', default=False, action='store_true',
+        help='If True, compress the json output.')
```

Comment:
It would be good to mention that the compression used is bzip2. I'm tempted to also say this should be called `--bzip2` because we may end up with a user on windows who doesn't have (and doesn't want to install) bzip2. They might ask us to add zip support...  anyway, I think `--compress` is fine for now. After all, it's probably our bzip2-capable code which is going to read this output anyway.